core/src/main/java/ai/timefold/solver/core/preview/api/domain/variable/declarative/ShadowVariableSessionFactory.java

```diff
@@ -0,0 +1,16 @@
+package ai.timefold.solver.core.preview.api.domain.variable.declarative;
+
+import ai.timefold.solver.core.impl.domain.solution.descriptor.SolutionDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.declarative.MockShadowVariableSessionFactory;
+
+import org.jspecify.annotations.NullMarked;
+
+@NullMarked
+public interface ShadowVariableSessionFactory {
```

Comment:
Maybe rename to something like ShadowVariableVerifier, as it serves a a simular purpose?

Reply:
Likewise, please remove from public API. The user has no use for this.